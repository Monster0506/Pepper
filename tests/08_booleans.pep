%% Testing boolean operators
LET x: int = 5
LET y: int = 5
LET z: int = 10
LET str1: string = "hello"
LET str2: string = "hello"
LET str3: string = "world"

%% Testing equality (&&)
SHOW("Testing equality (&&):")
SHOW(x && y)
SHOW(str1 && str2)

%% Testing inequality (&$$&)
SHOW("Testing inequality (&$$&):")
SHOW(x &$$& z)
SHOW(str1 &$$& str3)

%% Testing comparison operators
SHOW("Testing comparison operators:")
SHOW(x < z)
SHOW(z > x)
SHOW(x <= y)
SHOW(z >= x)

%% Testing logical operators (@$@, #$#, ~@)
SHOW("Testing logical operators:")
LET a: bool = true
LET b: bool = false

SHOW("a and b: a @$@ b:")
SHOW(a @$@ b)

SHOW("a or b: a #$# b:")
SHOW(a #$# b)

SHOW("not a: ~@ a:")
SHOW(~@ a)

SHOW("not b: ~@ b:")
SHOW(~@ b)

LET c: bool = x < z
LET d: bool = str1 && str2
LET e: bool = str1 &$$& str3
%% Testing complex conditions
SHOW("Testing complex conditions:")
SHOW(c @$@ d)
SHOW(~@ c #$# e)