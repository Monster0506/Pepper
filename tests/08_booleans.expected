Testing equality (&&):
true
true
Testing inequality (&$$&):
true
true
Testing comparison operators:
true
true
true
true
Testing logical operators:
a and b: a @$@ b:
false
a or b: a #$# b:
true
not a: ~@ a:
false
not b: ~@ b:
true
Testing complex conditions:
true
true